### ISSUE TITLE
Prototype builder method ProductDraftBuilder.of(ProductDraft) is lossy

### ISSUE THREAD

**Issue by selo-ecube** [commercetools/commercetools-jvm-sdk#1193]
http://commercetools.github.io/commercetools-jvm-sdk/apidocs/io/sphere/sdk/meta/ConstructionDocumentation.html#builders contains a section on prototype builders that suggests that e.g. [ProductDraftBuilder.of(ProductDraft)](https://github.com/commercetools/commercetools-jvm-sdk/blob/master/commercetools-models/src/main/java/io/sphere/sdk/products/ProductDraftBuilder.java#L44) creates a builder that can be used to obtain kind of a modified copy of the given product draft.

This is, however, not the case: https://github.com/commercetools/commercetools-jvm-sdk/blob/master/commercetools-models/src/main/java/io/sphere/sdk/products/ProductDraftBuilder.java#L44 has the following code:

```
public static ProductDraftBuilder of(final ProductDraft productDraft) {
        return of(productDraft.getProductType(), productDraft.getName(), productDraft.getSlug(), productDraft.getMasterVariant())
                .variants(productDraft.getVariants());
}
```
I.e. the implementation ignores e.g. the `publish` field.

**Comment by lauraluiz** [comment]
It is indeed a strange behaviour. In my opinion it should definitely return the builder with all the fields defined in the draft, not only the mandatory fields. @schleichardt what do you think?

**Comment by selo-ecube** [comment]
We've stumbled upon the issue again when we added categories to the `ProductsDraft` instances and it definitely feels like a bug.

I'd create a pull request but will need to do other things instead before I'm on vacation next week. We continue working around the issue.

Fixing the problem for `ProductDraftBuilder` would take less time than setting up a test that makes sure that the problem does not appear again and applying it for all builder methods that (should) have this prototype behaviour. The test could e.g. 

- create an input `ProductDraft`
- get the values of all (private) fields of the input `ProductDraft` (e.g. via `getClass().getDeclaredFields()` and `Field.setAccessible(true)` - maybe using recursion/iteration because fields might be inherited)
- perform outputProductDraft = ProductTypeBuilder.of(inputProductDraft).build()`
- get the values of all (private) fields of the output `ProductDraft`
- compare the values of the fields of input and output `ProductDraft`

The builder method could either be implemented like it is right now, or could maybe use reflection so it does not need to be adapted when further fields are added. Reflection comes with a runtime overhead, though. I'm not sure about your preferences here.
